torch_geometric/nn/models/lightgcn.py

```diff
@@ -0,0 +1,315 @@
+            sigmoid layer (:obj:`"ranking"`, :obj:`"link_prediction"`)
+            (default: :obj:`"ranking"`).
+        lambda_reg (int, optional): The :math:`L_2` regularization strength of
+            the Bayesian Personalized Ranking (BPR) loss for the "ranking"
+            objective (default: 1e-4).
+        alpha (float, optional): The vector specifying the layer weights in the
+            final embeddings. If set to :obj:`None`, the original paper's
+            uniform initialization of :obj:`(1 / num_layers + 1)` is used
+            (default: :obj:`None`).
+        **kwargs (optional): Additional arguments of the underlying
+            :class:`torch_geometric.nn.conv.MessagePassing` layers.
+    """
+
+    objectives = ['ranking', 'link_prediction']
+
+    def __init__(self, num_layers: int, num_nodes: int,
+                 embedding_dim: int = 32, objective: str = 'ranking',
+                 lambda_reg: float = 1e-4, alpha: Tensor = None, **kwargs):
+        super().__init__(**kwargs)
+
+        assert (num_layers >= 1), 'Number of layers must be larger than 0.'
+        assert (num_nodes >= 1), 'Number of nodes must be larger than 0.'
+        assert (embedding_dim >= 1), 'Embedding dimension must be larger ' \
+            'than 0.'
+        assert (objective in LightGCN.objectives), 'Objective must be one ' \
+            f'of {LightGCN.objectives}'
+
+        if alpha is not None:
+            assert (alpha.size(0) == (num_layers + 1)), 'alpha must be of ' \
+                'size (num_layers + 1).'
+        else:
+            alpha = [1 / (num_layers + 1)] * (num_layers + 1)
+
+        self.num_layers = num_layers
+        self.num_nodes = num_nodes
+        self.embedding_dim = embedding_dim
+        self.embeddings = Embedding(num_nodes, embedding_dim)
+        self.objective = objective
+        self.lambda_reg = lambda_reg
+        self.alpha = alpha
+        self.convs = ModuleList()
+
+        for _ in range(num_layers):
+            self.convs.append(LGConv())
+
+        if objective == LightGCN.objectives[0]:
+            self.loss_fn = BPRLoss(self.lambda_reg)
+        else:
+            self.loss_fn = BCELoss()
+
+    def reset_parameters(self):
+        self.embeddings.reset_parameters()
+
+    def forward(self, edge_index: Adj, edge_label_index: Adj = None,
+                weights: Tensor = None, **kwargs) -> Tensor:
+        """Computes rankings or link probabilities for pairs of nodes.
+
+        Args:
+            edge_index (Adj): Edge tensor specifying the connectivity of the
+                graph.
+            edge_label_index (Adj, optional): Edge tensor specifying the start
+                and end nodes for which to compute rankings or proabilities.
+                If :obj:`edge_label_index` is set to :obj:`None`, all edges in
+                :obj:`edge_index` will be used (default: :obj:`None`).
+            weights (Tensor, optional): Pre-computed embeddings or node
```

Comment:
I think this is pretty unusual and I prefer to remove this functionality. If the user wants to modify the initial embeddings, the user should do it directly by accessing `model.embedding`.